src/microbe_stage/OrganelleLayout.cs

```diff
@@ -234,4 +235,62 @@ IEnumerator IEnumerable.GetEnumerator()
     {
         return Organelles.GetEnumerator();
     }
+
+    /// <summary>
+    ///   Recursively loops though all hexes and checks if there any without connection to the rest.
+    /// </summary>
+    /// <returns>
+    ///   Returns a list of hexes that are not connected to the rest
+    /// </returns>
+    internal List<Hex> GetIslandHexes()
+    {
+        // The hex to start the recursion with
+        var initHex = Organelles[0].Position;
+
+        // These are the hexes have neighbours and aren't islands
+        var hexesWithNeighbours = new List<Hex> { initHex };
+
+        // These are all of the existing hexes, that if there are no islands will all be visited
+        var shouldBeVisited = Organelles.Select(p => p.Position).ToList();
+
+        CheckmarkNeighbors(hexesWithNeighbours);
+
+        // Return the difference of the lists (hexes that were not visited)
+        return shouldBeVisited.Except(hexesWithNeighbours).ToList();
+    }
+
+    /// <summary>
+    ///   A recursive function that adds the neighbours of current hex that contain organelles to the checked list and
+    ///   recurses to them to find more connected organelles
+    /// </summary>
+    /// <param name="checked">The list of already visited hexes. Will be filled up with found hexes.</param>
+    private void CheckmarkNeighbors(List<Hex> @checked)
+    {
+        while (true)
+        {
+            var endHexCount = 0;
+            foreach (var hex in @checked.ToList())
```

Comment:
I'm a bit suspicious about this, might also be related to the low performance.
Might something like the following work
```
var queue = new Stack type whatever it is<Hex>
queue.AddRange(@checked);

while(queue not empty)
{
                var hex = queue.pop;
                var myNeighbors = GetNeighborHexes(hex).Where(p => !@checked.Contains(p)).ToArray();
                if (myNeighbors.Length == 0)
                         continue;
                 queue.AddRange(myNeighbours);
                 @checked.AddRange(myNeighbours);
}
```
?

Reply:
I just copied Ices commit :P
I'll change it

Reply:
I was pretty sure that was the case. I didn't have the time to review that PR separately so I could only bring this suggestion up now.

Reply:
yeah, it drasicly improves #1794, but there is still much room for improvement